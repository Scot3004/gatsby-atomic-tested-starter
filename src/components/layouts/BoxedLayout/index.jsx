import React from 'react';
import PropTypes from 'prop-types';
import { Helmet } from 'react-helmet';
import { createGlobalStyle } from 'styled-components';
import Container from '../../atoms/Container';
import MainTitle from '../../atoms/MainTitle';
import MainContainer from '../../atoms/MainContainer';
import PageGrid from '../../atoms/PageGrid';

const GlobalStyle = createGlobalStyle`
  body {
    margin: 0;
    padding: 0;
    background: #f5f5f5;
  }
`;

export const BoxedLayout = ({ title, children }) => (
  <>
    <Helmet>
      <title>{title}</title>
    </Helmet>
    <GlobalStyle />
    <Container>
      <PageGrid>
        <header>
          <MainTitle>{title}</MainTitle>
        </header>
        <MainContainer>{children}</MainContainer>
      </PageGrid>
    </Container>
  </>
);

BoxedLayout.propTypes = {
  title: PropTypes.string.isRequired,
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node
  ]).isRequired
};

export default BoxedLayout;
