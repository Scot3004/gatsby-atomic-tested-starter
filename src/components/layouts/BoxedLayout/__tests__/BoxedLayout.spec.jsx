import React from 'react';
import 'jest-styled-components';
import renderer from 'react-test-renderer';
import BoxedLayout from '..';

test('it works', () => {
  const tree = renderer
    .create(<BoxedLayout title="title_stub">content stub</BoxedLayout>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
