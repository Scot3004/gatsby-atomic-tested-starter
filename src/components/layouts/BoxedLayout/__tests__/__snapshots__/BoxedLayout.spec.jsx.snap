// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it works 1`] = `
.c0 {
  display: grid;
  grid-template-columns: minmax(auto,64em);
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 1rem;
}

.c2 {
  color: #663399;
}

.c3 {
  min-width: 0;
}

.c1 {
  display: grid;
  grid-template-columns: 1fr;
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <header>
      <h1
        className="c2"
      >
        title_stub
      </h1>
    </header>
    <main
      className="c3"
    >
      content stub
    </main>
  </div>
</div>
`;
